### ISSUE TITLE
DefaultParams type doesn't count with optional parameters

### ISSUE THREAD

**Issue by Poky85** [molefrog/wouter#214]
Wouter export types:

```typescript
export interface DefaultParams {
  [paramName: string]: string;
}
export type Params<T extends DefaultParams = DefaultParams> = T;
```

Let's use dynamic segment in the path, eg:

```
<Route path="/:foo/:bar?">
  {(params) => {
    
  }}
</Route>;
```

...and match the route against path `/hello`. Then matcher returns params as following: `{ foo: hello, bar: undefined }`. But those params doesn't comply with `DefaultParams` definition. So I think it should be:

```typescript
export interface DefaultParams {
  [paramName: string]: string | undefined;
}
```

**Comment by molefrog** [comment]
Hey @itsMapleLeaf @cbbfcd could you take a look at the proposed change?

**Comment by cbbfcd** [comment]
@Poky85 may be resolved by #211 .

and @itsMapleLeaf, It looks like we need to add some test cases 😂
